test/unit/range/iterator_test_template.hpp

```diff
@@ -54,7 +54,7 @@ SEQAN3_CONCEPT has_expect_equal_member_function = requires(t & a)
 
 // Delegates to the test fixture member function `expect_eq` if available and falls back to EXPECT_EQ otherwise.
 template <typename T, typename A, typename B>
-void expext_eq(A && a, B && b)
+void expect_iter_value_equal(A && a, B && b)
```

Comment:
Just to make sure, did you read https://github.com/seqan/seqan3/pull/2004#discussion_r464267294?

Reply:
Nope, it was resolved. I don't go through resolved comments before I review. Thanks for pointing it out.

If I understood correctly, you want to change `expect_iter_equal` later on to also test if it points to the same position instead of just having the same value?

Note to the name before: `expect_eq` is the equivalent of google test `EXPECT_EQ` so I think the name is also spot-on, namely that I can expect that the two given values are equal? 
If you don't think so may I suggest `expect_value_equal` without the `iter`? because the function does not care whether it gets iter values or any other value. I was confused when I read it because to me it suggests that I need to pass an iterator.

Reply:
> because the function does not care whether it gets iter values or any other value.

That's not necessarily true, you can write functions that get unrelated data types and `expect_iter_value_equal` could make them comparable.

This would only work if we guarantee a particular argument order for `expect_iter_value_equal`. Now this argument order is dependent on the (unrelated) **iterator** values. This is the main reason why I called it `expect_iter_value_equal`, because it compares two (unrelated) iterator value. `expect_value_equal` would be in my eyes a bit weaker in its description, because it looses the aspect that it compares the values of the iterator.

I wouldn't even want to provide this functions if we wouldn't need it for the one use case, where we compare the values `compare(it[n], rng_it[n])`.

Reply:
Alright fine. Let's leave it and not spend to much time discussing it :)